flow/record/adapter/splunk.py

```diff
@@ -1,19 +1,42 @@
+import json
 import logging
 import socket
+import uuid
+from datetime import datetime, timezone
+from enum import Enum
+from typing import Optional
+from urllib.parse import urlparse
+
+try:
+    import requests
+    import urllib3
+
+    urllib3.disable_warnings()
+    HAS_REQUESTS = True
+except ImportError:
+    HAS_REQUESTS = False
 
 from flow.record.adapter import AbstractReader, AbstractWriter
+from flow.record.base import Record
+from flow.record.jsonpacker import JsonRecordPacker
 from flow.record.utils import to_base64, to_bytes, to_str
 
 __usage__ = """
 Splunk output adapter (writer only)
 ---
-Write usage: rdump -w splunk://[IP]:[PORT]?tag=[TAG]
+Write usage: rdump -w splunk+[PROTOCOL]://[IP]:[PORT]?tag=[TAG]&token=[TOKEN]&sourcetype=[SOURCETYPE]
```

Comment:
`[PROTOCOL]` is not documented here, i assume if it's not defined it will default to TCP.